.github/workflows/test-tutorials.yml

```diff
@@ -2,18 +2,38 @@ name: Test-Tutorials
 
 on:
   push:
+    branches:
+      - main
 
 jobs:
-  hello-world:
+  test-tutorials:
     runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        example-path:
+          - examples/zkapps/01-hello-world
+          - examples/zkapps/02-private-inputs-and-hash-functions
+          - examples/zkapps/05-common-types-and-functions
+          - examples/zkapps/10-account-updates
+
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v3
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
         with:
-          # In case of version change, update README.md accordingly
-          node-version: 18
-      - run: |
-          npm ci
+          node-version: '20'
```

Comment:
With the matrix strategy, we can run this on node 18, 20 and 22.  I think we might as well, so we know what works and doesn't work.

Reply:
It would be good to check, but do you think it’s worth the 3x time cost?

Reply:
Jobs that use matrix will be executed in parallel, so no extra cost.